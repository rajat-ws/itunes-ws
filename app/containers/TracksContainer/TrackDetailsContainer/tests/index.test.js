/**
 *
 * Tests for TrackDetailsContainer container
 *
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom';
import { renderProvider } from '@utils/testUtils';
import { TrackDetailsContainerTest as TrackDetailsContainer } from '../index';

describe('<TrackDetailsContainer /> container tests', () => {
  let trackDetails;
  let dispatchRequestTrackDetailsSpy;

  beforeEach(() => {
    trackDetails = {
      collectionName: 'collection name',
      previewUrl: 'https://audio-ssl.itunes.apple.com/itunes-assets/Music/43/63/eb/mzm.pmbvqxwh.aac.p.m4a',
      trackName: 'track name'
    };
    dispatchRequestTrackDetailsSpy = jest.fn();
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(
      <TrackDetailsContainer dispatchRequestTrackDetails={dispatchRequestTrackDetailsSpy} trackDetails={trackDetails} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
