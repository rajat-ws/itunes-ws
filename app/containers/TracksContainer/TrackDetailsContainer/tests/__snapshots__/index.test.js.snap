// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetailsContainer /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="TrackDetailsContainer__Wrapper-tevgmx-0 ENBzZ"
    >
      <div
        class="TrackDetailsContainer__TrackDetailsWrapper-tevgmx-1 jKPgDN"
      >
        <div
          class="ant-card ant-card-bordered TrackComponent__TrackCardContainer-sc-1y7fh5t-0 hZfrIv"
          data-testid="track-component"
        >
          <div
            class="ant-card-body"
          >
            <div
              class="ant-image"
            >
              <img
                class="ant-image-img"
              />
              <div
                class="ant-image-mask"
              >
                <div
                  class="ant-image-mask-info"
                >
                  <span
                    aria-label="eye"
                    class="anticon anticon-eye"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      />
                    </svg>
                  </span>
                  Preview
                </div>
              </div>
            </div>
            <div
              class="TrackComponent__StyledDescription-sc-1y7fh5t-1 ehlPZI"
            >
              <div
                class="ant-typography"
              >
                No artist name available
              </div>
              <div
                class="ant-typography"
              >
                <span
                  class="TrackComponent__StyledSpan-sc-1y7fh5t-2 hHSjOx"
                >
                   Collection name: 
                </span>
                 
                collection name
              </div>
              <div
                class="ant-typography"
              >
                <span
                  class="TrackComponent__StyledSpan-sc-1y7fh5t-2 hHSjOx"
                >
                   Track name: 
                </span>
                 
                track name
              </div>
              <div
                class="ant-typography"
              >
                <span
                  class="TrackComponent__StyledSpan-sc-1y7fh5t-2 hHSjOx"
                >
                   Country: 
                </span>
                 
              </div>
              <div
                class="ant-typography"
              >
                <span
                  class="TrackComponent__StyledSpan-sc-1y7fh5t-2 hHSjOx"
                >
                   Kind: 
                </span>
                 
              </div>
              <div
                class="ant-typography"
              >
                <span
                  class="TrackComponent__StyledSpan-sc-1y7fh5t-2 hHSjOx"
                >
                   Genre: 
                </span>
                 
              </div>
              <div
                class="ant-typography"
              >
                <span
                  class="TrackComponent__StyledSpan-sc-1y7fh5t-2 hHSjOx"
                >
                   Wrapper Type: 
                </span>
                 
              </div>
            </div>
            <div
              class="TrackComponent__ButtonWrapper-sc-1y7fh5t-5 dehKAG"
            >
              <button
                class="TrackComponent__PlayTrackBtn-sc-1y7fh5t-3 khRFCr"
              >
                <span
                  class="TrackComponent__ButtonLabel-sc-1y7fh5t-6 fSyCiY"
                >
                   Play 
                </span>
              </button>
            </div>
            <audio
              data-testid="trackAudio"
              src="https://audio-ssl.itunes.apple.com/itunes-assets/Music/43/63/eb/mzm.pmbvqxwh.aac.p.m4a"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
